compiler/rustc_passes/src/check_attr.rs

```diff
@@ -270,6 +270,16 @@ impl CheckAttrVisitor<'tcx> {
                                 )
                                 .emit();
                         }
+                        if attr.style == AttrStyle::Inner && CRATE_HIR_ID == hir_id {
+                            self.tcx
+                                .sess
+                                .struct_span_err(
+                                    meta.span(),
+                                    "`#![doc(alias = \"...\")]` isn't allowed as a crate \
+                                     level attribute",
+                                )
+                                .emit();
```

Comment:
```suggestion
                                .emit();
                            return false;
```

If we emit an error then we should return `false` like the other attribute checks do.